src/AutoRest.CSharp/Common/Generation/Types/CSharpType.cs

```diff
@@ -597,7 +597,7 @@ public static CSharpType FromLiteral(CSharpType type, object literalValue)
         /// <param name="unionItemTypes">The list of union item types.</param>
         /// <param name="isNullable">Flag used to determine if a type is nullable.</param>
         /// <returns>A <see cref="CSharpType"/> instance representing those unioned types.</returns>
-        public static CSharpType FromUnion(IReadOnlyList<CSharpType> unionItemTypes, bool isNullable)
+        public static CSharpType FromUnion(IReadOnlyList<CSharpType> unionItemTypes, bool isNullable = false)
```

Comment:
it seems like we should remove `isNullable` from public constructor, we never set the nullability from public users

Reply:
well the idea of keeping this is for future usage to be parallel with the ctor of this type